tools/rules-check/rules-check.cc

```diff
@@ -50,6 +51,7 @@ int main(int argc, char **argv) {
         const char *arg = *args;
         std::string err;
         int r;
+        bool need_free = false;
```

Comment:
How about making args a std::string?

Reply:
Do you mean `arg` (and not `args`)? Because it makes sense to me, but after a quick review I don't understand the idea.

Reply:
Okay I'm done. See the result after my last commit [cdb6b839](https://github.com/owasp-modsecurity/ModSecurity/pull/3114/commits/cdb6b839e1de54ce126a2cc8e97a9f2c2846b5a9)

Reply:
Please ignore the new SonarCloud [issue](https://sonarcloud.io/project/issues?resolved=false&pullRequest=3114&id=owasp-modsecurity_ModSecurity&open=AY6AlC0CBkWqhBwO48MX) - that's not because of this PR.